From 6a4ff4d30899898b8ee2576f93342bc8193f3b07 Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <richard.a.griffiths@intel.com>
Date: Tue, 2 Feb 2016 14:57:12 -0800
Subject: [PATCH 6/6] [EM/PMIC_CCSM]: add bounds check in debug

Correct a possible boundary overflow by replacing sprintf
with snprintf.

Change-Id: I1c62cd6f2d15b7c0caa507342e4eca70ca11f6bb
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18665
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/465719
---
 drivers/power/intel_pmic_ccsm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 10dab55..e9c2c5e 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -487,7 +487,7 @@ static void pmic_debugfs_init(void)
 	reg = (u16 *)chc.reg_map;
 	for (i = 0; i < chc.reg_cnt; i++, reg++) {
 
-		sprintf(name, "%s", pmic_regs_name[i]);
+		snprintf(name, sizeof(name), "%s", pmic_regs_name[i]);
 
 		fentry = debugfs_create_file(name,
 				S_IRUGO,
@@ -508,7 +508,7 @@ static void pmic_debugfs_init(void)
 
 	for (i = 0; i < pmic_tt_reg_cnt; i++) {
 
-		sprintf(name, "%s", pmic_tt_regs[i].reg_name);
+		snprintf(name, sizeof(name), "%s", pmic_tt_regs[i].reg_name);
 
 		fentry = debugfs_create_file(name,
 				S_IRUGO,
-- 
1.9.1

